### ISSUE TITLE
Fixing the Ranking Process System by adding another dedicated team

### ISSUE THREAD

**Issue by Arkener1** [Quaver/Quaver#3313]
### Introduction

(You can skip this part, since it's not relevant to the issue, but it should be relevant for the people who wants some veracity on some crucial statements on this request).

Hello, i'm Arkener, i've been playing VSRGs almost 10 years ago (mostly Stepmania, osu!mania). After the years, by playing thousands and thousands of maps, i got enough experience to analyze maps good enough to help mappers by modding (but i want to clarify that i don't consider myself a mapper, since I don't have enough patience to actually sit for hours and make something good).

After entering to Quaver, I realized that there was a LOT of maps and mappers that had the potential to improve, so i applied to the RS team (to keep in touch with the actual mappers and collaborate to help the community) and after a couple of months, i left due to several reasons (time, differences of opinion between other RS members, the whole RS system, etc).

Nowadays, there's a lot of issues with the ranking process, so i came up with the following idea:

### The actual problem

There's plenty of maps nowadays that aren't checked good enough by the RS team, mostly due to the fact that is not their job to improve the map's quality, and that causes the overall quality of the ranked maps to decrease significantly. Sometimes is offset, sometimes is just the way it's mapped (no music representation, uncomfortable pattern choices, unjustified notes/unbalance that affect directly the gameplay, etc).

And of course, none of those points exists on the requirements of getting a map ranked, but those are the points that **_players_** look at to tell whether a map is good or not. Sadly, since the Ranking Criteria doesn't specify anything about mapping, **_it's impossible to new mappers to tell if they're going a good job or no, so they just have to move on until their maps starts getting ranked._** Same applies to RS when they have to judge the mapping to accept it or not, because since not everyone is experienced in some mapping aspects, they tend to Accept/Deny according to their own judgement, opinion and perceptions; and this point is crucial since the maps that gets ranked sometimes are contradictory according to the points that different RS members evaluate (some prioritize overall gameplay, specific things about mapping without covering the whole map, whether it's "funny" or not, mapping consistency, etc).

Obviously, the modding page always existed to help with the tedious work of giving proper feedback to encourage mappers to improve - but it's a double-edged sword because everyone can comment on it. Instead of getting good feedback, they could risk the possibility of making the map worse.

Currently, the modding page is not commonly used for feedback purposes (some common issues are: not everyone explains why something doesn't work; why it should be changed and how; and there are people that don't even know how to give proper feedback.). But we can start from there to encourage mappers and modders to improve the maps that are getting ranked and also to motivate the Quaver community to collaborate and improve this system!

### The solution

The first stage should be adding little changes to the structure of the ranking process. Currently the overall process more or less consists of:

1. Map gets submitted to the ranking queue.
2. RS checks metadata/other stuff, depending of the RS.
3. RS decide to whether Accept, Deny or Hold, depending on the case.

The issue starts pretty much at point 2. My solution consists of adding another team and making an extra step in the process that takes place before the RS check. Let's call this team "Mapping Supervisors" (or MS to abbreviate). It's their job to help make sure the map is good enough to be ranked, and it would work like this:

1. Map gets submitted to the ranking queue.
2. MS checks the map and looks for anything regarding mapping issues, including giving proper mods to help mappers know what to fix, why to do it, and how.
3. MS marks the map as Accepted or Denied to be judged by the RS team (if it's denied, then it gets a little bit of cooldown outside of the ranking queue, specifying why it was denied and what to fix to get it accepted next time).
4. RS checks metadata and/or technical stuff (bpm/unsnapped notes, etc).
5. RS decide to whether Accept, Deny or Hold (depending on the case).

**_Also to make this work, the requirement of 2 RS to rank a map should change to 1 MS + 1 RS in order to make it efficient, since RS won't be touching anything regarding mapping unless it's something more technical._**

### But hey, shouldn't be this slower since there's more steps and also more work involved?

Not necessarily. The slow speed of maps getting ranked is mainly an issue because maps aren't good enough to be ranked - but they're also not bad enough to be denied. Because the RS don't mod; this causes an issue where the map gets trapped in the queue until someone decides to Accept or Deny the map. The idea of the MS is to ensure that this situation does not arise.

### Where would you get the people for that "Mapping Supervisors" team?

Similar to applying for RS, there should be a MS application and the main MS Leader could check it to start the growth of the MS team. At the same time, this gives the guarantee that the members are capable of doing what they're supposed to do.

### This is a lot of work, how can you encourage the community to join and do something like that?

Of course giving any kind of reward might result in people doing it only for the purpose of the reward - so it's important that we should not go too far and give too much as some kind of incentive. Some reasonable incentives could include:

- Color names
- Titles on player's profile (something like "Featured Mapper" or a badge proving the continuous support over the time).
- Profile Customization (this involves a lot of work, so i would mention it only as a random idea).

_This idea could tie into the proposal about adding more customizable and cosmetic features to Quaver._

### By giving people privileges like these, you are actually causing a divide in the community with this "team"

Because the MS team is supposed to be the ones who are in charge of modding, there should be a system too about voting mods from other players; acknowledging their hard work, and giving them the possibility to get the same rewards, so everyone can be at the same boat!

### I want those kind of rewards but i don't like modding

Obviously this is the first step of getting new features for players - so in the future, they might be able to get rewards like that by doing other activities!

### Other alternatives/ideas

The easiest solution would be encouraging people to do more mods - but as I said before, not every instance of feedback is good feedback. We should regulate this and have some kind of heirarchy to mark mods as "mods from the MS team, mods acknowledged by the MS team as good, and normal mods", so mappers can tell the difference without getting confused about people giving different mods on the same sections of the map and give priority to the best feedback.

The other solution is to reform the whole Ranking Criteria so the RS can focus on what the actual requirements to rank a map are; but that would also mean that they would have to give more mods than usual (because they would be looking more at the mapping persepctive and that's not their job).

Because of what is mentioned above, this also implies that with the integration of the team, there should be some kind of "rulebook" to follow - so every MS can fix any mapping issue (something like "Mapping Criteria").

### What's the difference between this and osu!mania, we're not making the same thing, right?

Absolutely not. in this system we're going to split the ranking process into two different areas: the mapping area and the metadata area. They're gonna work together, and players that doesn't belong to the team can still collaborate to rank the maps since they're able to give mods to maps that are in the ranking queue, and in the long term, MS will be always growing, nominating potential modders that helped a lot in modding pages and acknowledging other mods, giving to mappers the guarantee that those mods are good enough to improve the quality.

### So... What's the Mapping Supervisor's job then?

- Check the maps on the Ranking Queue.
- Giving mapping feedback to improve the map according to "Mapping Criteria".
- Accept/Deny/Hold the map so RS can check it afterwards and rank the map.
- Acknowledging potential modders that can help on the ranking process (maybe adding a system of voting from MS so the mapper can say "oh, this mod it's not from a MS but it's voted by them, so it's good to check).

### Additional info

I personally think that this could be very helpful since this game is still new. There's lots of new players and mappers, and having this kind of system (that would lead maybe to future kinds of rewards by helping to improve the game in any way) can motivate a lot of people from the community. Of course this idea can be improved on, but I hope it gets noticed enough to fix some of the issues mentioned above!

**Comment by nebneb0703** [comment]
I feel like creating a whole new team with an arbitrary name to perform the same job is unnecessary, the fundamental aspects are still the same.

I do believe that a Mapping Criteria would be good, both for ensuring consistency between supervisors, and for beginner mappers who want to look for some guidance.

However, by creating a new team, you're not solving the problem, just moving it elsewhere.

Modding a map and giving feedback is a time consuming process. But by having a separate team deal with that, and then pass it to the other team achieves nothing. MS would be doing all the hard work, whereas RS would simply be verifying the tags. Furthermore, a lot of "technical" stuff as you describe to remain as the RS's job, could just as easily be picked up by MS, for example an invalid bpm. If you have an invalid bpm, then your mapping needs fixing. 

You still have a team of people who are modding maps for an extended time, which causes the queue to slow down and pile up, whereas the RS queue would be empty most of the time, since checking metadata and technical stuff for a map that has *already been verified as "good quality" by the votes of MS* wouldn't take nearly as much time as modding. This disparity means that RS would have a pretty boring and effortless job, and most people who applied for RS now would probably have preferred to be "MS", since they joined based on those hopes, similar to your backstory.

I suggest extending the RS team so that more people can work in parallel through the ranking queue. Even in an MS approach, that team would still have to be large enough to sieve through the queue at a reasonable speed, and simultaneously provide good quality feedback. With the current system, you can accept more RS. 

Furthermore, ensure that RS give good quality feedback and mods. This can be made easier for the RS with a descriptive Marking Criteria. There is no reason to split this into two separate roles, and it would not improve efficiency.

**Comment by Arkener1** [comment]
> I feel like creating a whole new team with an arbitrary name to perform the same job is unnecessary, the fundamental aspects are still the same.

Well, the current job of RS is only checking metadata (or critical issues like offset/bpm/unsnapped notes, but there's cases that it's not even noticed), and people who wants to check mapping (almost no one) are completely voluntary. The idea is to make a team with people who has the ability to check the actual mapping, and that's not Ranking Supervisor's job.
 
> However, by creating a new team, you're not solving the problem, just moving it elsewhere.

Moving it to an area where there's people capable to solve it, as mentioned earlier.

> Modding a map and giving feedback is a time consuming process. But by having a separate team deal with that, and then pass it to the other team achieves nothing. MS would be doing all the hard work, whereas RS would simply be verifying the tags. Furthermore, a lot of "technical" stuff as you describe to remain as the RS's job, could just as easily be picked up by MS, for example an invalid bpm. If you have an invalid bpm, then your mapping needs fixing.

Same as mentioned before, we need people to do the work that is not being done; the average RS only checks whether it's "good or not" (judging only by their own preferences), offset, tags, source (this is metadata), and then Accepts if it's decent, Hold if there's some small detail that is easy to fix (but this is not always correct, since they even have denied maps by an incorrect source on the metadata), and Deny if there's too many to fix.

Also, I mentioned the technical stuff because not every RS checks just metadata, there's people who checks other things, but as i mentioned before on the request "not every instance of feedback is good feedback.". Not every RS has mapping knowledge, and there's no rulebook to follow the same idea.

> You still have a team of people who are modding maps for an extended time, which causes the queue to slow down and pile up, whereas the RS queue would be empty most of the time, since checking metadata and technical stuff for a map that has _already been verified as "good quality" by the votes of MS_ wouldn't take nearly as much time as modding. This disparity means that RS would have a pretty boring and effortless job, and most people who applied for RS now would probably have preferred to be "MS", since they joined based on those hopes, similar to your backstory.

Exactly. When i was RS and proposed the idea (obviously it wasn't that structured like now, but it still consisted on a separated team, everyone refused because they don't wanted to mod anything; only 2-3 people said that they would support by joining as "MS". It's very time-consuming, but there's also the factor of "MS voting player's mods", so other players can help to wipe the ranking queue by modding the maps; this is why i mentioned the possibility of a reward to encourage this kind of support.

> I suggest extending the RS team so that more people can work in parallel through the ranking queue. Even in an MS approach, that team would still have to be large enough to sieve through the queue at a reasonable speed, and simultaneously provide good quality feedback. With the current system, you can accept more RS.

Personally, i completely reject this lol. There's a lot of RS that differs from opinion and sometimes that was the reason why some maps had 1/2 votes eternally; with a good "Mapping Criteria" this **could** be fixed, but again, as mentioned before, not everyone has a lot of mapping knowledge to judge correctly.

> Furthermore, ensure that RS give good quality feedback and mods. This can be made easier for the RS with a descriptive Marking Criteria. There is no reason to split this into two separate roles, and it would not improve efficiency.

Not RS's job. To be honest, when i joined i expected this to be a rule, but it wasn't; maybe you can make them to give good feedback (explaining why the hold/denied status), but everything that regards modding, it's not and it will never be their job. Even if it becomes a rule, some would leave, and the people forced not necessarily could do a good job since they're not that experimented, unlike other people who are specialized at modding.

**Comment by nebneb0703** [comment]
Okay so I'll start off by saying that I obviously do not have a personal experience with being RS or interacting with RS. You are an ex-RS, you saw how the system worked from the inside, and you do have better and more accurate perspective.

Gonna tackle these out of order since it makes more sense to structure my reply in that way.

> everyone refused because they don't wanted to mod anything; only 2-3 people said that they would support by joining as "MS". It's very time-consuming

I assumed that people who apply to RS (from my present perspective, might have changed since then idk) do so because they enjoy mapping and wish to help charters get rankable (and then ranked) maps. To be honest, I assumed that ***the job*** was much more similar to your proposition of MS. Now I'm confused, is the job literally to do a brief check then press accept/deny/hold? If there are members of the RS team who accept this responsibility, but not modding (and being brutally honest here) maybe they shouldn't really be entrusted to judge maps and cast an honest vote. At this point, I would suggest *only* having an MS team, since the few additional responsibilities that an RS would have could also be done by the same people. 

From what I've seen (though not paying significant enough attention, do expect discrepancies), it seems that the RS focus on getting through the queue as fast as possible, and then modding just becomes out of the question. This is obviously not helpful, hence your suggestion, but I also don't see why a separate team would help with this issue. I think reformation of the current team would be adequate here, since I don't see a need for multiple teams.

> Personally, i completely reject this lol. There's a lot of RS that differs from opinion and sometimes that was the reason why some maps had 1/2 votes eternally; with a good "Mapping Criteria" this could be fixed, but again, as mentioned before, not everyone has a lot of mapping knowledge to judge correctly.

I would assume that in your suggested MS team, the members would have adequate mapping knowledge to provide quality feedback to charters. Why can this not be applied to the current RS team? If you cannot give a quality judgement outside of "ew/pog", then I don't think you should be in charge of ranking maps (I do not intend to suggest that current RS rank based on ew/pog lol. I'm going along with the assumption in the argument).

There could be other systems added to ensure that maps don't end up in limbo. For example, charts with already 1/2 votes will get moved to a higher priority. Obviously, if an RS was working on a different map at the time, its alright to continue, the main idea is that maps with 1/2 votes are not ignored, and maybe even have a separate page to highlight them. 

Following along from my point of introducing more RS to the team, switching from 2 votes to 3 votes where you need 2 deny votes to completely reject a map may allow RS to be more critical of each other, and therefore reach common ground, hopefully lowering the significance the problem of differences in opinion. Ideally, this would promote more discussion amongst the RS to challenge each other's views, and actually collaborate as a team, rather than individually trying to get through the queue as fast as possible with lots of denies.

> Not RS's job. To be honest, when i joined i expected this to be a rule, but it wasn't; maybe you can make them to give good feedback (explaining why the hold/denied status), but everything that regards modding, it's not and it will never be their job. Even if it becomes a rule, some would leave, and the people forced not necessarily could do a good job since they're not that experimented, unlike other people who are specialized at modding.

Well you're complaining that the RS team isn't good enough (in regards to modding), and therefore a new team is required, but then your point here is "oh no but they'll leave the team!". I know this is probably brutal thinking again, but RS *should* have high responsibility. It should come with rigorous requirements. If the requirements now change so that the job of RS is to provide good quality feedback and modding to charts, and people leave or do not do an adequate job, then they are not fit for this role. If they do not satisfy your needs and therefore a new "replacement" team is necessary (though you imply that they'll work alongside each other, but my point still remains that they should be the same team), then there is an inconsistency in your arguments. But I'm optimistic and I don't think this is as big of an issue as you make it out to be. I have trust in some of the current RS that I know. I know that some other users, maybe even yourself, would like to join RS if it meant helping actually mod.

> Moving it to an area where there's people capable to solve it, as mentioned earlier.

Small thing to note for this, but it should be obvious by the points I have made prior. We both agree that there needs to be people specialised to solve this. I believe it should be a reformation of the current team. A new MS team would render the RS team useless imo, and I wouldn't like to accept that they have the final say despite not having a thorough and honest analysis. I believe that would make it unfair for users. At the same time, if the RS job in this two-team system is simply metadata and small technical issues, then deny isn't really necessary anymore, and the problem I just described wouldn't exist, but then the need for the RS team to be separate also wouldn't exist, since these would be so small, the MS team could just account for that. Now that RS is not necessary, and effectively replaced by MS, there is no reason to have a split in the first place. And as I already described, I have optimism that a reform could work.

**Comment by Arkener1** [comment]
> is the job literally to do a brief check then press accept/deny/hold? If there are members of the RS team who accept this responsibility, but not modding (and being brutally honest here) maybe they shouldn't really be entrusted to judge maps and cast an honest vote.

Absolutely yes. This is why this request is oriented to give the biggest responsibility to a specialized team that knows how to mod.

>it seems that the RS focus on getting through the queue as fast as possible, and then modding just becomes out of the question. This is obviously not helpful, hence your suggestion, but I also don't see why a separate team would help with this issue. I think reformation of the current team would be adequate here, since I don't see a need for multiple teams.

They try, obviously because i'm no longer RS and the team drastically changed (and also there's no queue from maps submitted like 5+ months ago), i guess the situation is better now, but there's still the issue that modding is not a must-to-do thing. A separate team would totally help because they will be doing what RS aren't doing, basically. A reformation of the team may sound the best option, but it's not the idea to go and say "okay everyone leaves", _**because there should be at least a leader that has the enough knowledge to guide everyone, make such reforms, designing the Mapping criteria, and making grow the team with also experienced modders, and well... If there was someone like that on the team, everything would be fixed already.**_

>I would assume that in your suggested MS team, the members would have adequate mapping knowledge to provide quality feedback to charters. Why can this not be applied to the current RS team?

I will be honest here, when i joined, there was like 8 RS (i think more, maybe they were afk, idk), only 3 of them gave good feedback, others checked judging by their personal preferences and just voted, there was a lot of mistakes made because of this (just to give an example, this map that was insta ranked without even realizing that the audio started like 300ms later lol https://quavergame.com/mapset/3065 ). Nowadays, those 3 doesn't belong to the team anymore, and many mappers joined. Like i said before, maybe the situation improved, but it's better to give people the role that they fit in the best, and not every mapper knows how to mod correctly (if that were the case, there would be a lot of perfect maps made at the 1st try, and that doesn't exist).

TL;DR, in the past, RS were not experienced at modding because they didn't know/they didn't wanted to/they joined only for voting. Now, you just need to check the last ranked maps, check them and realize that there's almost no mods involved since the quality sometimes is even worse (https://quavergame.com/mapset/8911 for example). We need a fresh team, with a good leader, and good members at what they're doing.

>There could be other systems added to ensure that maps don't end up in limbo. For example, charts with already 1/2 votes will get moved to a higher priority.

With or without the idea, this should be already a rule, since the normal priority is the most decent map that is been found, or mappers that are known to give decent quality (enough to insta-vote).

>Following along from my point of introducing more RS to the team, switching from 2 votes to 3 votes where you need 2 deny votes to completely reject a map may allow RS to be more critical of each other.

Not gonna lie, when they switched from 1 vote to 2, the only thing it changed was sending the link saying "checked, pls vote", adding another vote needed won't change anything, since it's hard to check a map right next to other person: when you're checking metadata, that's a task where you only need 1 person to check. And when you're modding, unless the other person has the exact same thoughts with the exact same neutrality as the other person, it won't work. 

>Well you're complaining that the RS team isn't good enough (in regards to modding), and therefore a new team is required, but then your point here is "oh no but they'll leave the team!". I know this is probably brutal thinking again, but RS should have high responsibility.

Sadly there's a lot of people that contributed to the ranking process, and the proposal of make them leave will end in an absolutely denial.

>If they do not satisfy your needs and therefore a new "replacement" team is necessary (though you imply that they'll work alongside each other, but my point still remains that they should be the same team), then there is an inconsistency in your arguments. 

In the request i mentioned that it should be something like 1 MS + 1 RS accepting the map in order to get it ranked. Also i think i already mentioned several times what the team is lacking, so can you please elaborate where's the inconsistency?

> I have trust in some of the current RS that I know. I know that some other users, maybe even yourself, would like to join RS if it meant helping actually mod.

This may fix the issue, but as i said before, there's gonna be several probelms:

1. Some RS will leave (since not everyone would want to invest time in modding)
2. There's people who won't be expelled since they already collaborated to the game, and not everyone has the enough knowledge to mod correctly
3. Adding the mod rule to RS would imply that the current head leader should be good enough to elaborate the Mapping Criteria, coordinate everyone who wants to know more about modding, expell the people who won't help and add people with the potential of doing a good job at modding. and like i said above, "If there was someone like that on the team, everything would be fixed already.".

>A new MS team would render the RS team useless imo

Not necessarily, there's always a margin of human error, and someone who does a second check is always needed

>At the same time, if the RS job in this two-team system is simply metadata and small technical issues, then deny isn't really necessary anymore, and the problem I just described wouldn't exist, but then the need for the RS team to be separate also wouldn't exist, since these would be so small, the MS team could just account for that.

Deny exists with the only purpose of pulling away a map that require tons and tons of fixes in order to be ranked; something like "almost everything is wrong because ____. after evaluating that and fixing it, please submit it again". And by mixing people with different responsibilities would also mean that those responsibilities will be mixed as well, since both sides are going to argue if one side doesn't like the other's opinion. (this happened a lot in the past, even while being one single team).

>And as I already described, I have optimism that a reform could work.

A reform could only work if Swan does it by itself honestly. And this is 100% my personal opinion, but optimism doesn't guarantee results; i would prefer to be sure to fix a problem rather than hope to be fixed in the future.

Overall, i think that a lot of these decisions that we're talking about should only be taken by him, because the impact of things like reforming a whole team or adding another rulebook to change all ranked maps requirements, is way too big for a single faction of the game, so let's hope and wait for his answer before discussing more about crucial decisions.

**Comment by Skalim112** [comment]
Let me contribute with my experience on RS.

i joined the team because i wanted to bring feedback to the maps and mappers, and because i liked the fact that you have direct contact with the mappers on discord. Once i ended the trial, i was evaluated like an annoying modder for writing a lot of threads with suggerences on the maps and for denying maps "with ranking potential" so, they suggested me to mod less and vote more. That means that modding is not the priority of the RS team, the priority is just "checking" if the map is following the ranking criteria, which means quality over quantity besides of how the map is made (and nowadays quality is almost minimal)

Some mappers did actually improved his mapping style due to our contribution on the team (and also because some mappers just talked with us asking for tips for improve), meanwhile some RS just speedrunned the voting/ranking check because "i liked the map", after all, the activity inside the team is all that matters (i want to be wrong if this is happening nowadays), influencing considerably the quality of the maps.

Also, the forgotten maps on the pending queue happens because nobody wants to check the patterning of the map, mainly because the map is actually bad, and when you don´t have a justification in mind about what to say to the mapper, that map goes to the abyss, which means that the RS team don´t have enough experience for judge patterns, making responses like "send this map to a friend for help you" and not giving the proper feedback to the mapper.

If you don´t trust me, check the Ranking Process: 
![brUh](https://user-images.githubusercontent.com/58799659/143176948-8993c721-4992-4856-bd97-d862f112b531.png)

**Comment by Illuminati-CRAZ** [comment]
idea: change vote/deny system to approve/neutral/disapprove system

approve is +1 points, neutral 0, disapprove -1

rank at 2 points, auto-deny after some criteria such as a week passing since first vote or 5 votes or smtg

**Comment by Adrriii** [comment]
If we don't mention the work needed to put a new team system in place, bringing changes to the ranking queue etc ...

The principle of a modding team is good, although in the mindset of Quaver's ranking it is not a mandatory process. We need to keep the process as-is in the cases where it does work. That is why I think we need another approach than what you proposed, to include this extra team in the process, where it will really be needed (because I feel like it is needed, as you point it).

My proposal is that, when the RS is checking the map, if it is not in a rankable state due to quality of the mapping, then it could be moved to the queue checked by MS where they can decide whether to take the map on their shoulders and mod it, or put the final bullet and completely deny it. This seems like a very smooth process where both teams play a crucial role, and have way clearer implications than in the current state of the ranking queue.

**Comment by AiAe** [comment]
> If we don't mention the work needed to put a new team system in place, bringing changes to the ranking queue etc ...
> 
> The principle of a modding team is good, although in the mindset of Quaver's ranking it is not a mandatory process. We need to keep the process as-is in the cases where it does work. That is why I think we need another approach than what you proposed, to include this extra team in the process, where it will really be needed (because I feel like it is needed, as you point it).
> 
> My proposal is that, when the RS is checking the map, if it is not in a rankable state due to quality of the mapping, then it could be moved to the queue checked by MS where they can decide whether to take the map on their shoulders and mod it, or put the final bullet and completely deny it. This seems like a very smooth process where both teams play a crucial role, and have way clearer implications than in the current state of the ranking queue.

To add to this idea since its what I think its more reasonable:
There should be a mapper consent for his map to be modded, if they approve then MS can go to the queue and move if they wish to be also improved. 
That does not mean RS won't be able to send the map to be checked by MS.

Also if MS mods a map they can also give a vote after they think its ready. 
The possible ranking will be RS+RS or MS+RS.

**Comment by Jolomann** [comment]
Mapper consent is a great idea. Many denied maps are left to rot by the creators, and sending everything to MS could lead to wasted effort

**Comment by Arkener1** [comment]
>My proposal is that, when the RS is checking the map, if it is not in a rankable state due to quality of the mapping, then it could be moved to the queue checked by MS where they can decide whether to take the map on their shoulders and mod it, or put the final bullet and completely deny it.

lmao, i explicitly said that not every RS's judgement is good enough to tell whether something is quality or not, i sent 2 maps to at least give an idea, the reason behind it, my experience of RS and even Skalim said it as well. Nothing will change if RS take the decisions; the "quality filter" has to happen before, with people who can see what can be polished and what not.

**Comment by Arkener1** [comment]
>There should be a mapper consent for his map to be modded, if they approve then MS can go to the queue and move if they wish to be also improved.

You're right, but slightly wrong, because there's unexperienced mappers who really needs some modding to make them realize what they have to improve/fix/change. Also about consent, this is already a thing and you can mark the mods as Accepted/Denied/Ignored, and in the case it's denied/ignored, as long as the mapper gives a justified reason, it should be okay on that case.

And yeah, i said that the possible best case for ranking process with the idea should be 1MS + 1RS vote

**Comment by Arkener1** [comment]
>Mapper consent is a great idea. Many denied maps are left to rot by the creators, and sending everything to MS could lead to wasted effort

The old method that i used to do when i was modding some maps, was to check the mapper's activity, because normally if you submit for ranked, in the 99% of cases they will listen to you, so they can rank the map. There has been cases where people left the game at the same month the map was submitted, and they were pretty much ignored (but it was also funny since they were active on discord).

So there should be a better approach, something like on mania, when  you submit it as Work in Progress, or just as pending. The bad thing about that is not everybody knows if their map is completely correct for ranking, so at least there should be in the submitting process, some checkmark like "Accepting modding" or something like that, to guarantee that MS work won't be in vain.

**Comment by Arkener1** [comment]
Okay, after talking with @AiAe on how to make this idea work on the long-term, and also how to start implementing it, this is gonna be the plan of the **_first stage_**:

1. Discuss with the current RS team so we can all rework the Ranking Criteria (a rework that it was always necessary)
2. Implement the new team little by little, starting from the ranking requirement of 1 MS + 1 RS. meanwhile RS are checking one side of the map, MS would work only on the mapping aspects, so they kinda work together to rank the map.
3. With the continuous help of the MS team, we're gonna standarize and document the basics of a future Mapping Criteria, so new mappers can check this rulebook to know how to submit a rankable map.

Future steps like adding the features i mentioned on the main request (like MS voting for potential good mods, rewards for supporting this whole mod thing and so on), will be discussed more in detail after everything of this is already done and works properly.
